profile/src/main/java/rdx/works/profile/data/repository/AccessFactorSourceProvider.kt

```diff
@@ -0,0 +1,42 @@
+package rdx.works.profile.data.repository
+
+import com.radixdlt.extensions.removeLeadingZero
+import kotlinx.coroutines.flow.first
+import rdx.works.profile.data.model.compressedPublicKey
+import rdx.works.profile.data.model.extensions.nextAccountIndex
+import rdx.works.profile.data.model.factorsources.DerivationPathScheme
+import rdx.works.profile.data.model.factorsources.DeviceFactorSource
+import rdx.works.profile.data.model.factorsources.FactorSource
+import rdx.works.profile.data.model.pernetwork.DerivationPath
+import rdx.works.profile.derivation.model.KeyType
+import rdx.works.profile.derivation.model.NetworkId
+import javax.inject.Inject
+
+class AccessFactorSourceProvider @Inject constructor(
```

Comment:
I am a bit confused with the purpose of this class. It does not access any factor sources, it looks like to me a wrapper of methods that will get increasingly bigger in the future like a utils class. Both methods below have no connection together.

Reply:
my initial idea is to push all the related to access factor sources (either derive public key or signing) logic in the profile module and keep the viewmodel of the bottomsheet with the minimum logic. And then this "provider" would return the result (public key, signers) to the UI.
However, that's not really possible because some factor sources e.g. ledger are off device. 

But I still think to proceed with this way and see how this will escalate. Maybe this provider can be a "AccessOnDeviceFactorSourcesWhatever".

This refactoring is definitely not in the final version( in this PR) and it's in my mind that design can change later when implementing the rest of factor sources.

Reply:
also keep in mind that this class would go away with the shared lib 😄

Reply:
Understood, I get your point